File: remote/src/main/java/com/kelsos/mbrc/mappers/CoverMapper.java
package com.kelsos.mbrc.mappers;

import com.kelsos.mbrc.dao.CoverDao;
import com.kelsos.mbrc.dto.library.CoverDto;
import java.util.List;
import rx.Observable;

public class CoverMapper {
  public static List<CoverDao> map(List<CoverDto> objects) {
    return Observable.from(objects).map(CoverMapper::map).toList().toBlocking().first();
  }

  public static CoverDao map(CoverDto object) {
    CoverDao dao = new CoverDao();
    dao.setId(object.getId());
    dao.setHash(object.getHash());
    dao.setDateUpdated(object.getDateUpdated());
    dao.setDateDeleted(object.getDateDeleted());
    dao.setDateAdded(object.getDateAdded());
    return dao;
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/mappers/CoverMapper.kt
package com.kelsos.mbrc.mappers

import com.kelsos.mbrc.dao.CoverDao
import com.kelsos.mbrc.dto.library.CoverDto

object CoverMapper {
  fun map(objects: List<CoverDto>): List<CoverDao> {
    return objects.map { map(it) }.toList()
  }

  fun map(item: CoverDto): CoverDao {
    val dao = CoverDao()
    dao.id = item.id
    dao.hash = item.hash
    dao.dateUpdated = item.dateUpdated
    dao.dateDeleted = item.dateDeleted
    dao.dateAdded = item.dateAdded
    return dao
  }
}
